src/common/util.h

```diff
@@ -36,7 +36,12 @@
 #define __attribute__nonnull__(x) assert((x) != NULL)
 #endif
 
-/*
+#ifdef _MSC_VER
+#define OSCAP_HIDDEN_START
```

Comment:
when we now have OSCAP_API does it even make sense to keep these macros? can't we just replace everything with OSCAP_API? If there is no OSCAP_API the symbol would be invisible.

Reply:
Great idea! We can do it according to https://gcc.gnu.org/wiki/Visibility

However, on GCC we now use `_Pragma("GCC visibility push(hidden)")` , and the article says:

> #pragma GCC visibility is stronger than -fvisibility; it affects extern declarations as well. -fvisibility only affects definitions, so that existing code can be recompiled with minimal changes. This is more true for C than C++; C++ interfaces tend use classes, which are affected by -fvisibility. 

Do you think it is a problem?

Reply:
Yes, it is. My suggestion is to get rid of OSCAP_HIDDEN_START everywhere. You can use OSCAP_API even for GCC to make symbols visible.

Reply:
@mpreisler Yes, I understand that. I meant: Do you think that what the quotation from the article says is a problem?